packages/neptune-extension/src/common/core/configuration.js

```diff
@@ -0,0 +1,42 @@
+import {
+  getAccessToken,
+  getNotebookData,
+} from 'common/api/backend';
+
+export function validateConfiguration(metadata, token) {
+  if (!metadata.notebookId) {
+    // Notebook without neptune id - yet no setup.
+    return Promise.reject();
+  }
+
+  if (!token) {
+    // Token was deleted - do the setup again.
+    return Promise.reject();
+  }
+
+  return new Promise((resolve, reject) => {
```

Comment:
1. You don't need to wrap this code in another Promise.

You can just write this:

```js
        return getAccessToken(token)
            .then(({ data: authData }) => {

                return getNotebookData(metadata.notebookId)
                    .then(({ data: notebookData }) => {

                        if (authData.username !== notebookData.owner) {
                            // username changed.
                            Promise.reject();
                        }

                        // Configuration ok!
                        return Promise.resolve();
                    })
            });
```

2. You could simplify this and other codes that communicates with API if you would decide to use central point for updating the token. So that code making requests to API doesn't have to handle it.

For instance you can use axios interceptors: https://github.com/axios/axios#interceptors

If the the updating token would be in a single place then we can simplify the code to just only:

```js
                return getNotebookData(metadata.notebookId)
                    .then(({ data: notebookData, authData }) => {

                        if (authData.username !== notebookData.owner) {
                            // username changed.
                            Promise.reject();
                        }

                        // Configuration ok!
                        return Promise.resolve();
                    })
```

Reply:
I wonder how it came to this that I missed that part of spec. This seems like much a better way, in that sense that you do not have to handle reject in all nested promises.